src/main/java/com/thelow_quest_helper/thelow_quest_helper/item/NextLocationBeacon.java

```diff
@@ -0,0 +1,292 @@
+        finalDestinationZ = finalDestZ;
+        currentNextDistName = initialNextDistName;
+
+        try {
+            int[] nextDistCoords = parseCoordinates(initialNextDistCoordsStr);
+            currentNextDistX = nextDistCoords[0];
+            currentNextDistY = nextDistCoords[1];
+            currentNextDistZ = nextDistCoords[2];
+
+            int[] nextGoalCoords = parseCoordinates(initialNextGoalCoordsStr);
+            currentNextGoalX = nextGoalCoords[0];
+            currentNextGoalY = nextGoalCoords[1];
+            currentNextGoalZ = nextGoalCoords[2];
+
+        } catch (IllegalArgumentException e) {
+            System.err.println("[NextLocationBeacon] 初期座標のパースエラー: " + e.getMessage());
+            setToggleBeacon(false); // エラー時はトグルオフ
+            return;
+        }
+
+        double distToFinalFromInitialNextGoal = calculateDistanceSimple(currentNextGoalX, currentNextGoalY, currentNextGoalZ, finalDestinationX, finalDestinationY, finalDestinationZ);
+        boolean alreadyAtFinalByInitialGoal = (distToFinalFromInitialNextGoal <= REACH_THRESHOLD);
+        
+        if (currentNextDistName.equals("目的地") && alreadyAtFinalByInitialGoal) {
+            System.out.println("[NextLocationBeacon] 最初の案内が既に最終目的地です。");
+            EntityPlayer player = Minecraft.getMinecraft().thePlayer;
+            if (player != null) {
+                 double playerDistToFinal = calculateDistanceSimple(player.posX, player.posY, player.posZ, finalDestinationX, finalDestinationY, finalDestinationZ);
+                 if (playerDistToFinal <= REACH_THRESHOLD) {
+                    System.out.println("[NextLocationBeacon] プレイヤーは既に最終目的地にいます。案内を終了します。");
+                    setToggleBeacon(false);
+                    return;
+                 }
+            }
+        }
+        
+        setToggleBeacon(true);
+        guidingPlayer = true;
+        commandSendTimer = 0;
+        activateBeacon();
+        System.out.println("[NextLocationBeacon] 案内開始。最終目的地: (" + finalDestX + "," + finalDestY + "," + finalDestZ + ")");
+    }
+
+    /**
+     * 現在の nextDist にビーコンを表示（または更新）します。
+     */
+    private static void activateBeacon() {
+        if (guidingPlayer && toggleBeacon) {
+            System.out.println("[NextLocationBeacon] ビーコン表示: " + currentNextDistName +
+                               " @ (" + currentNextDistX + "," + currentNextDistY + "," + currentNextDistZ + "). 次の目標: ("+currentNextGoalX+","+currentNextGoalY+","+currentNextGoalZ+").");
+            // 実際のBeaconBeamDrawerクラスのメソッドを呼び出す
+            // 例: com.Sura.Matatabi.events.BeaconBeamDrawer.displayPoint(currentNextDistX, currentNextDistY, currentNextDistZ, currentNextDistName);
+            // もし events.BeaconBeamDrawer が別パッケージなら、適切なインポートと呼び出しが必要です。
+            // ここでは仮に events.BeaconBeamDrawer.displayPoint としています。
+            com.thelow_quest_helper.thelow_quest_helper.item.BeaconBeamDrawer.displayPoint(currentNextDistX, currentNextDistY, currentNextDistZ, currentNextDistName);
```

Comment:
MarkerRendererに渡すように変更してみて様子見をしたいです
```suggestion
            MarkerRenderer.addMarker((double)currentNextDistX, (double)currentNextDistY, (double)currentNextDistZ, currentNextDistName);
```